### DISCUSSION TITLE
post/7/

### DISCUSSION THREAD

**Discussion by giscus** [HoshinoRei/blog#7] [General]
# post/7/

小玲有这样一个需求，一个无线路由器能发出多个 SSID 的 WiFi，并且连接一个 SSID 的 WiFi 的设备与连接另一个 SSID 的 WiFi 的设备之间的通信可以隔离。这里的隔离不是

https://blog.bling.moe/post/7/

**Comment by Senorsen** [comment]
感谢分享，看完一遍正准备实现一下。
~另外按照 [User scripts 文档](https://github.com/RMerl/asuswrt-merlin.ng/wiki/User-scripts) ，还可以在 wan-event - connected 和 nat-start 里也加上试试，以前用 R6300V2 + merlin 时，有个 vlan 的脚本放在这两个钩子下，就算改了路由器配置也不会被覆盖。~
👆 my suggestion above is not working on AX86U Pro

**Comment by miudin** [comment]
你好，目前我把路由器改AP模式了，想把WAN口改为Trunk口，wan tag vlan 2 用作iptv，lan1 untag vlan2. 其它接口依然是vlan 1该如何操作呢？麻烦指导一下，谢谢~

**Reply by HoshinoRei** [reply]
你好。理论上可以用这样的命令，实际可不可以不知道呢。

```bash
brctl delif br0 eth4
brctl addbr br1
ip link set br1 up
brctl addif br1 eth0.2 eth4
```

**Comment by miudin** [comment]
您好，首先感谢您的回复。看了您的教程后，突然想起还能用ip link来设vlan。本人自测以下命令可以实现iptv IPoE透传拨号。上联是交换机在16口设为trunk口，默认vlan1，光猫itv口接交换机设untag vlan2



ip link add link eth0 name eth0.2 type vlan id 2

ip link set eth0.2 up
 
brctl addbr br1
 
ip link set br1 up
 
brctl delif br0 eth1
 
brctl addif br1 eth0.2 eth1





Haifeng
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "HoshinoRei/blog"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年5月27日(星期六) 凌晨0:32
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [HoshinoRei/blog] post/7/ (Discussion #7)





 
你好。理论上可以用这样的命令，实际可不可以不知道呢。
 brctl delif br0 eth4 brctl addbr br1 ip link set br1 up brctl addif br1 eth0.2 eth4

**Comment by JanydChang** [comment]
你的方案完美符合我的需求，但是我没有Debian做路由。请问ros路由器能实现你这个DHCP服务器根据不同接口分配不同段IP的功能吗？

**Comment by miudin** [comment]
可以

**Reply by JanydChang** [reply]
方便给个教程或者关键词吗😊不知道如何描述这个需求

**Comment by aaronzgh** [comment]
很棒的帖子，感谢博主。总在折腾服务器，家里还几个，但是之前搞网络方面的极少，有个相关的问题请教一下。
我也是 AX86U，因为服务器最近想部署 K8S + Ceph 集群，了解到想建立 VLAN，准备顺便把家里物联网设备以及可能的访客网络都一起配置了。
目前我的网络拓扑是 光猫（桥接） 到 AX86U主路由（拨号+DHCP 等），然后后面有线无线直接或通过交换机间接，来连接服务器及其他家庭设备。
其中有个 HPE microserver 小服务器，在上面跑了 openwrt（具体是 immortalwrt）软路由，在旁路网关的角色，来实现一些额外的网络功能。有需要的设备就将其网络的网关配置指向软路由。这种旁路网关的好处是，如果软路由出问题了，直接把家庭设备的网关重新指向 AX86U 即可，不会大面积影响家人上网。
看博主的 AX86U 是配置成 AP模式，不知道我这种拓扑，是不是区别不大。

**Reply by miudin** [reply]
OP更多是路由的扩展，例如多拨，多WAN，还有一些不可描述的功能。如果没这些需求更建议直接用AX86U做主路由，毕竟这些硬路由有专门的NAT加速，小包转发更好。游戏这些小包的延迟更低。

**Reply by aaronzgh** [reply]
嗯，主路由是直接用的 AX86U，主要就是留学，然后搞 smartdns、WireGuard 之类的，放在 OP 了，在旁路网关的角色，所以出问题也不影响一般正常上网。
所以看博主是以软路由做主路由，AX86U 配置成 AP 角色，才有上面的问题。

**Reply by miudin** [reply]
我跟楼主是一样的，AX86U用来做AP，OP做主路由。
我的宽带可以8拨，AX86U满足不了我。

**Reply by HoshinoRei** [reply]
区别还是挺大的吧，用旁路由之后 IPv6 和 DHCP 之间有点问题。设置静态 IP 的话，因为 IPv6 地址经常变，也不现实。所以小玲还是觉得如果需要透明代理的话，软路由作为主路由最靠谱。

**Reply by aaronzgh** [reply]
非常感谢！
1. 用旁路由之后，旁路由配置的关闭 DHCP、IPv6 功能，仍然会有问题吗？不知道博主指的什么问题？
2. 静态 IP 应该可以 IPV4 IPV6 单独设置，如果配置 IPV4静态、IPV6动态 不知道是否可以？

另外，还有一个跳出这个具体问题的，我折腾服务器和码代码的时候会觉得很自由，但是网络问题就总会束手束脚的，因为可能过程中让家里都无法上网，尤其是博主安排软路由作为主路由的情况。
不知道日常的测试，有什么软件工具或者什么样的架构安排（比如我把 OP 安排在旁路主要就是这个考虑，还建立了两个 软路由VM，一个测试挂了如果想快速恢复可以直接指向备用 OP），能够很方便快捷地进行测试，如果大面积影响家庭网络又能快速恢复的一些思路吗

**Reply by HoshinoRei** [reply]
1. 小玲指的是主路由可以通过 DHCP 让 IPv4 的网关指向旁路由，但是 IPv6 就不行了。如果全都不用 IPv6，那就不用担心这个问题。
2. 可以，但是用透明代理，IPv6 的网关又没指向旁路由的话（使用动态不好指向旁路由），IPv6 的流量就不能被透明代理了。

小玲的方法是用备用软路由。小玲有一个 R4S 和一个 R2S，主要是使用 R4S。如果 R4S 玩崩了，就拔下来换上 R2S。R4S 修好后再换上 R4S。

**Comment by ivan509-hub** [comment]
这个教程完美地符合我的需求，设备也差不多，不过我的是R2S（openwrt，目前也是旁路由结构）。openwrt也可以这样设置吗？

**Reply by HoshinoRei** [reply]
当然可以了，其实我现在已经换成 Openwrt 了。
